linera-base/src/crypto/secp256k1.rs

```diff
@@ -119,30 +163,29 @@ impl FromStr for Secp256k1PublicKey {
     type Err = CryptoError;
 
     fn from_str(s: &str) -> Result<Self, Self::Err> {
-        let pk = secp256k1::PublicKey::from_str(s).map_err(CryptoError::Secp256k1Error)?;
-        Ok(Secp256k1PublicKey(pk))
+        hex::decode(s)?.as_slice().try_into()
     }
 }
 
 impl TryFrom<&[u8]> for Secp256k1PublicKey {
     type Error = CryptoError;
 
     fn try_from(value: &[u8]) -> Result<Self, Self::Error> {
-        let pk = secp256k1::PublicKey::from_slice(value).map_err(CryptoError::Secp256k1Error)?;
-        Ok(Secp256k1PublicKey(pk))
+        Self::from_bytes(value)
     }
 }
 
 impl fmt::Display for Secp256k1PublicKey {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        let s = hex::encode(self.0.serialize());
-        write!(f, "{}", s)
+        let str = hex::encode(self.0.to_encoded_point(true).to_bytes());
+        write!(f, "{}", str)
```

Comment:
Would `str.fmt(f)` respect formatting options (e.g. `format!("{:8}", pub_key)`)?

Reply:
My version was slightly more performant since we were cutting of the unneeded bytes but this is more elegant.

Reply:
(I don't understand where we cut off the bytes? Anyway, just a thought; I'm fine with either variant, of course!)

Reply:
Not here ofc but in the Debug variants.

Reply:
Ah, good point, I totally didn't think of that! Feel free to revert, if you prefer.

And if you don't: I also missed that 9 bytes is 18 characters, not 9 characters. :facepalm: